API/Controllers/OPDSController.cs

```diff
@@ -580,19 +583,23 @@ private static UserParams GetUserParams(int pageNumber)
     public async Task<IActionResult> GetReadingListItems(int readingListId, string apiKey, [FromQuery] int pageNumber = 0)
     {
         var userId = await GetUser(apiKey);
+
         if (!(await _unitOfWork.SettingsRepository.GetSettingsDtoAsync()).EnableOpds)
             return BadRequest(await _localizationService.Translate(userId, "opds-disabled"));
-        var (baseUrl, prefix) = await GetPrefix();
+
         var user = await _unitOfWork.UserRepository.GetUserByIdAsync(userId);
+        if (user == null)
+            return Unauthorized();
+
+        var readingLists = await _unitOfWork.ReadingListRepository.GetReadingListDtosForUserAsync(user.Id, true, GetUserParams(pageNumber), false);
+        if (readingLists == null)
+            return Unauthorized();
 
-        var userWithLists = await _unitOfWork.UserRepository.GetUserByUsernameAsync(user!.UserName!, AppUserIncludes.ReadingListsWithItems);
-        if (userWithLists == null) return Unauthorized();
-        var readingList = userWithLists.ReadingLists.SingleOrDefault(t => t.Id == readingListId);
+        var readingList = readingLists.SingleOrDefault(rl => rl.Id == readingListId);
         if (readingList == null)
-        {
             return BadRequest(await _localizationService.Translate(userId, "reading-list-restricted"));
```

Comment:
Same comment about the if statement, since this is long, it makes sense to keep the {}